lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
Could you explain why we need to pass the current `FileView` instance to the file icons service? Maybe it's not a huge deal, but seems like the service shouldn't need it in order to provide an icon. What am I missing? 💭